packages/cheerio-crawler/src/internals/cheerio-crawler.ts

```diff
@@ -794,16 +793,19 @@ export class CheerioCrawler<JSONData = unknown> extends BasicCrawler<
 
         const gotOptions = {} as OptionsInit;
 
-        if (this.useSessionPool) {
-            this._applySessionCookie(crawlingContext, gotOptions);
-        }
-
         const { request, session } = crawlingContext;
-        const cookieSnapshot = request.headers?.Cookie ?? request.headers?.cookie;
+        const preNavigationHooksCookies = request.headers?.Cookie ?? request.headers?.cookie ?? '';
```

Comment:
it feels like we are introducing quite a lot of code duplication, could we try to move at least parts of this to basic crawler?

Reply:
We kind of can't... each extended crawler overrides how the hooks are called / where they're called, so we need to have this. We could, however, make the `request.headers?.Cookie ?? request.headers?.cookie ?? ''` a basic crawler method to reduce that code logic dupe

Reply:
Nevermind, remembered what I wanted to change and now code dupe is no more